src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryStepTest.java

```diff
@@ -85,7 +85,11 @@ public class LibraryStepTest {
             null, null, Collections.<GitSCMExtension>emptyList())));
         s.setChangelog(false);
         r.assertEqualDataBoundBeans(s, stepTester.configRoundTrip(s));
-        snippetizerTester.assertRoundTrip(s, "library changelog: false, identifier: 'foo@master', retriever: legacySCM([$class: 'GitSCM', branches: [[name: '${library.foo.version}']], extensions: [], userRemoteConfigs: [[url: 'https://nowhere.net/']]])");
+        if (r.jenkins.get().getPlugin("git").getWrapper().getVersion().startsWith("4.")) {
```

Comment:
OK for now (a cleanup PR in a few weeks would be appreciated) though in general we prefer to avoid this kind of switch—rather just update the test dependency to the newer version. If you have not released that version yet, file a draft PR with an incremental version so it is ready to go and be included in an omnibus `bom` PR.

Reply:
Will do.